app/controllers/concerns/defence_request_authorization.rb

```diff
@@ -1,19 +1,24 @@
 module DefenceRequestAuthorization
 
+  private
+
   def find_defence_request
+    defence_request_id = params[:defence_request_id] || params[:id]
     @defence_request = DefenceRequest.find(defence_request_id)
   end
 
-  def set_policy_with_context
-    @policy ||= policy(PolicyContext.new(defence_request, current_user))
-  end
-
   def defence_request
     @defence_request ||= DefenceRequest.new
   end
 
   def new_defence_request_form
```

Comment:
This doesn't have anything to do with authorisation, should it be somewhere else?